cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -320,6 +385,143 @@ const char* trim_utf8(gdv_int64 context, const char* data, gdv_int32 data_len,
   return data + start;
 }
 
+// Trims characters present in the trim text from the left end of the base text
+FORCE_INLINE
+const char* ltrim_utf8_utf8(gdv_int64 context, const char* basetext,
+                            gdv_int32 basetext_len, const char* trimtext,
+                            gdv_int32 trimtext_len, int32_t* out_len) {
```

Comment:
The utf8 handling seems incorrect. You need to decode the utf8 char and match against the target string instead of matching individual bytes. In this case a byte of multibyte char in trim_text might match against a byte of another multibyte char in target string